internal/ethapi/api.go

```diff
@@ -1643,9 +1648,8 @@ func newRPCTransaction(tx *types.Transaction, blockHash common.Hash, blockNumber
 		result.GasFeeCap = (*hexutil.Big)(tx.GasFeeCap())
 		result.GasTipCap = (*hexutil.Big)(tx.GasTipCap())
 		// if the transaction has been mined, compute the effective gas price
-		if baseFee != nil && blockHash != (common.Hash{}) {
-			// price = min(gasTipCap + baseFee, gasFeeCap)
-			result.GasPrice = (*hexutil.Big)(effectiveGasPrice(tx, baseFee))
+		if receipt != nil {
```

Comment:
I'm not sure if this is equivalent to the code before. When called from `NewRPCPendingTransaction` my understanding is that the GasPrice will change with this changes for a `DynamicFeeTxType`.

Reply:
`NewRPCPendingTransaction` calls `newRPCTransaction` with `blockHash = common.Hash{}`, so the if would not trigger in the old version. Neither would it in the old version because `receipt = nil`. So I don't see what difference it would make for `NewRPCPendingTransaction` calls.

That does not mean that I'm confident that the change won't have unintended side effects elsewhere. But I could not spot any and I did not want to leave one more case in the code for which I can't see the use.

Reply:
> calls newRPCTransaction with blockHash = common.Hash{}, so the if would not trigger in the old version

Good point, I missed that.